src/features/History/historySlice.ts

```diff
@@ -109,4 +110,14 @@ startHistoryListening({
   },
 });
 
+startHistoryListening({
+  actionCreator: restoreChat,
+  effect: (action, listenerApi) => {
```

Comment:
Is this when restoring from history or the cache?

Also thread is in the `restoreChat` action, so it would be best to use that as middleware seems to be called before the reducers.

Reply:
I think both. The reason I added it is because otherwise the "read" flag wouldn't get updated in the history.

So the read flag gets set to true when the chat is done streaming and being opened from the history. And this code is there to save that.

Maybe it can be done in a better way?

Reply:
you can check the restore payload, that'll have more reliable data for the thread and if it's been read or not :)